src/System/Taffybar/Widget/Workspaces.hs

```diff
@@ -700,9 +705,9 @@ defaultGetWindowIconPixbuf =
 
 unscaledDefaultGetWindowIconPixbuf :: WindowIconPixbufGetter
 unscaledDefaultGetWindowIconPixbuf =
-  getWindowIconPixbufFromDesktopEntry <|||>
-  getWindowIconPixbufFromClass <|||>
-  getWindowIconPixbufFromEWMH
+  handleIconGetterException getWindowIconPixbufFromDesktopEntry <|||>
```

Comment:
@kurnevsky just thinking out loud here, but might it be possible to apply this automatically as part of <|||> or something?

Reply:
If not, I think I would prefer it if these were part of the definition of `getWindowIconPixbufFromDesktopEntry` etc. instead of it only being applied to the default.

Reply:
also, especially in the case of `getWindowIconPixbufFromDesktopEntry` I think we should log a warning that even though there was an icon to apply for the window, we could not apply it because the icon was missing from the theme.

Without something like that, it would likely be difficult for people to work out that their icon theme is not working for some reason.

Reply:
> as part of <|||>

In this case it won't work with a single getter. I can apply it to the definition of these functions though.

Reply:
> In this case it won't work with a single getter. I can apply it to the definition of these functions though.

Yeah i think maybe this is the best course of action. I think with scaledPixbufGetter and now this, there are just too many random wrapping functions to add to these pixbuf getters, and that stuff is likely going to confuse most users.

Reply:
> In this case it won't work with a single getter. I can apply it to the definition of these functions though.

well im not saying erase the existing function, just apply it automatically as part of <|||>

Reply:
But then it won't work if user defines single getter with no usages of `<|||>`.

Reply:
Instead I just applied it to each of these getters.